### PR TITLE
Fix rpaths patcher being run on symbolic links

### PR DESCRIPTION
**Pull request by mbargull**
### Description

Use `skip_symlinks=True` for former `is_codefile`/`codefile_type` calls.
This just mechanically adds `skip_symlinks=True` to the calls sides that previously had this implicitly (before https://github.com/conda/conda-build/commit/1dc24708cae103bd1c3913520df91bb53de17c87).

Fixes gh-5179 .

### Checklist - did you ...

- [x] Add a file to the `news` directory ([using the template](../blob/main/news/TEMPLATE)) for the next release's release notes?
     
- [x] Add / update necessary tests?

### CODE REVIEW

**Comment by mbargull** [comment]
This currently targets `3.28.x` since `24.1` is not yet used in conda-forge.
We should do one of these:
1. Create new patch releases for `3.28`/`24.1`,
2. Create a patch release for `24.1` and use build/use that package on conda-forge.
3. Add https://github.com/conda/conda-build/pull/5181.patch as a patch in conda-forge/conda-build-feedstock and build a `3.28.4.post1` in case we can't do `3.28` release and can't use `24.1` in production yet.

This bug is a bit problematic since it doesn't cause builds to outright fail, but rather leads to silently create partially broken packages..

File: tests/test_post.py [hunk 1]
```diff
@@ -148,3 +154,26 @@ def test_menuinst_validation_fails_bad_json(testing_config, caplog, tmp_path):
...
+        f"conda_build.post.{func_name}",
+        side_effect=getattr(post, func_name),
+    )
+    api.build(
+        os.path.join(metadata_dir, "_rpath_symlink"),
+        config=testing_config,
+        variants={
+            "rpaths_patcher": ["patchelf", "LIEF"],
+            **variants_conda_build_sysroot,
+        },
```
Comment (mbargull) [inline tests/test_post.py:175]:
```suggestion
def test_rpath_symlink(mocker, testing_config):
    if on_linux:
        func_name = "mk_relative_linux"
    elif on_mac:
        func_name = "mk_relative_osx"
    mk_relative = mocker.patch(
        f"conda_build.post.{func_name}",
        side_effect=getattr(post, func_name),
    )
    api.build(
        os.path.join(metadata_dir, "_rpath_symlink"),
        config=testing_config,
        variants={"rpaths_patcher": ["patchelf", "LIEF"]},
```

**Review by kenodegard — COMMENTED** [review]
wow this was sneaky, I went back through the old PRs and agree that these are the old `is_codefile`/`codefile_type` calls

I'm in favor of doing another `24.1` patch, I'd prefer not to do another `3.28` unless we absolutely have to

File: tests/test_post.py [hunk 2]
```diff
@@ -148,3 +154,23 @@ def test_menuinst_validation_fails_bad_json(testing_config, caplog, tmp_path):
...
+@pytest.mark.skipif(on_win, reason="rpath fixup not done on Windows.")
+def test_rpath_symlink(mocker, testing_config):
+    if on_linux:
+        func_name = "mk_relative_linux"
+    elif on_mac:
+        func_name = "mk_relative_osx"
+    mk_relative = mocker.patch(
+        f"conda_build.post.{func_name}",
+        side_effect=getattr(post, func_name),
+    )
```
[the remark this answers is not in the document]
Comment (mbargull) [reply tests/test_post.py]: Ah, didn't see that one! Nice, thanks.

**Comment by mbargull** [comment]
> wow this was sneaky

Indeed -- kind of fortunate that we ran into the `patchelf` bug to notice it :).

> I'm in favor of doing another `24.1` patch, I'd prefer not to do another `3.28` unless we absolutely have to

How about we merge this, release `24.1.2` and build it on conda-forge today and only if something else breaks, we look at `3.28` again?

**Review by kenodegard — APPROVED** [review]

**Pull request merged**